capstone/src/test/java/com/google/sps/DatabaseServiceTest.java

```diff
@@ -0,0 +1,121 @@
+        entity.setProperty(Form.URL_PROPERTY_KEY, DatabaseServiceTest.TEST_URL);
+        DatabaseService.save(entity);
+
+        Query query = new Query(Lesson.LESSON_ENTITY_NAME);
+        PreparedQuery results = datastore.prepare(query);
+
+        try{ 
+            Lesson lesson = DatabaseService.getLesson(results.asSingleEntity().getKey());
+
+            Assert.assertEquals(lesson.getType(), Lesson.TYPE_FORM);
+            Assert.assertEquals(lesson.getTitle(), DatabaseServiceTest.TEST_TITLE);
+            Assert.assertEquals(lesson.getDescription(), DatabaseServiceTest.TEST_DESCRIPTION);
+            Assert.assertTrue(lesson instanceof Form);
+            Form form = (Form) lesson; 
+            Assert.assertEquals(form.getEditUrl(), DatabaseServiceTest.TEST_EDIT_URL);
+            Assert.assertEquals(form.getUrl(), DatabaseServiceTest.TEST_URL);
+
+        } catch (EntityNotFoundException e) {
+            Assert.fail("EntityNotFoundException Thrown");
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void serializeJsonMissingTypeTest() {
+        Entity entity = new Entity(Lesson.LESSON_ENTITY_NAME);
+        entity.setProperty(Lesson.TITLE_PROPERTY_KEY, DatabaseServiceTest.TEST_TITLE);
+        entity.setProperty(Lesson.DESCRIPTION_PROPERTY_KEY, DatabaseServiceTest.TEST_DESCRIPTION);
+        entity.setProperty(Form.EDIT_URL_PROPERTY_KEY, DatabaseServiceTest.TEST_EDIT_URL);
+        entity.setProperty(Form.URL_PROPERTY_KEY, DatabaseServiceTest.TEST_URL);
+        DatabaseService.save(entity);
+        Query query = new Query(Lesson.LESSON_ENTITY_NAME);
+        PreparedQuery results = datastore.prepare(query);
+
+        try{ 
+            Lesson lesson = DatabaseService.getLesson(results.asSingleEntity().getKey());
+        } catch (EntityNotFoundException e) {
+            Assert.fail("EntityNotFoundException Thrown");
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void serializeJsonWrongTypeTest() {
+        Entity entity = new Entity(Lesson.LESSON_ENTITY_NAME);
+        entity.setProperty(Lesson.TYPE_PROPERTY_KEY, "MalformedForm");
+        entity.setProperty(Lesson.TITLE_PROPERTY_KEY, DatabaseServiceTest.TEST_TITLE);
+        entity.setProperty(Lesson.DESCRIPTION_PROPERTY_KEY, DatabaseServiceTest.TEST_DESCRIPTION);
+        entity.setProperty(Form.EDIT_URL_PROPERTY_KEY, DatabaseServiceTest.TEST_EDIT_URL);
+        entity.setProperty(Form.URL_PROPERTY_KEY, DatabaseServiceTest.TEST_URL);
+        DatabaseService.save(entity);
+        Query query = new Query(Lesson.LESSON_ENTITY_NAME);
+        PreparedQuery results = datastore.prepare(query);
+
+        try{ 
+            Lesson lesson = DatabaseService.getLesson(results.asSingleEntity().getKey());
+        } catch (EntityNotFoundException e) {
+            Assert.fail("EntityNotFoundException Thrown");
+        }
+    }
```

Comment:
Decided to leave the other tests for each type since I didn't want to let this PR be big. I could add it in a future PR along with servlet testing (I got it to work I think).